pkg/kubelet/config/config.go

```diff
@@ -293,21 +295,24 @@ func (s *podStorage) seenSources(sources ...string) bool {
 func filterInvalidPods(pods []api.BoundPod, source string) (filtered []*api.BoundPod) {
 	names := util.StringSet{}
 	for i := range pods {
-		var errors []error
-		name := podUniqueName(&pods[i])
+		pod := &pods[i]
+		var errlist []error
+		name := podUniqueName(pod)
 		if names.Has(name) {
-			errors = append(errors, apierrs.NewFieldDuplicate("name", pods[i].Name))
+			errlist = append(errlist, apierrs.NewFieldDuplicate("name", pod.Name))
 		} else {
 			names.Insert(name)
 		}
-		if errs := validation.ValidateBoundPod(&pods[i]); len(errs) != 0 {
-			errors = append(errors, errs...)
+		if errs := validation.ValidateBoundPod(pod); len(errs) != 0 {
+			errlist = append(errlist, errs...)
 		}
-		if len(errors) > 0 {
-			glog.Warningf("Pod %d (%s) from %s failed validation, ignoring: %v", i+1, pods[i].Name, source, errors)
+		if len(errlist) > 0 {
+			glog.Warningf("Pod %d (%s) from %s failed validation, ignoring: %v", i+1, pod.Name, source, errlist)
+			//FIXME: is it kosher to log an event about a pod that might be invalid?
```

Comment:
Yes, this is a great idea. Otherwise users will get no feedback when kubelet ignores their pods.